swim/ping_handler.go

```diff
@@ -34,6 +37,18 @@ func handlePing(node *Node, req *ping) (*ping, error) {
 		Changes: req.Changes,
 	})
 
+	if req.App == "" {
+		if node.requiresAppInPing {
+			node.logger.Warnf("Rejected ping from unknown ringpop app %s", req.App)
+			return nil, errors.New("Pinged ringpop requires app name")
+		}
+	} else {
+		if req.App != node.app {
+			node.logger.Warnf("Rejected ping from wrong ringpop app %s", req.App)
```

Comment:
normally we would use something like `node.logger.WithField("app", req.App).Warn("Rejected ping from wrong ringpop app")`. That way searching in ES is easier because all the messages are the same.